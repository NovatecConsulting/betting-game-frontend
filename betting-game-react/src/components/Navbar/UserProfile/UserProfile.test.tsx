import React from "react";
import { shallow } from "enzyme";
import { UserProfile } from ".";
import { Profile } from "oidc-client";

describe(UserProfile.name + " component", () => {
    afterEach(() => {
        jest.clearAllMocks();
    });
    it("should render and match snapshot with given profile.", () => {
        const profileMock = {
            name: "TestName",
            email: "test@test.com",
            picture: "test.png",
        } as Profile;

        const wrapper = shallow(<UserProfile profile={profileMock} />);

        expect(wrapper.find("div").at(1)).toIncludeText("TestName");
        expect(wrapper.find("img").prop("src")).toEqual("test.png");
        expect(wrapper.find("img").prop("alt")).toEqual("test@test.com");
        expect(wrapper).toMatchSnapshot();
    });
});
