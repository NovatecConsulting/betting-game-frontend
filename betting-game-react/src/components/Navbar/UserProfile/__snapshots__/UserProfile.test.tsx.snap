// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile component should render and match snapshot with given profile. 1`] = `
<div
  className="flex items-center justify-between flex-wrap "
>
  <img
    alt="test@test.com"
    height={40}
    src="test.png"
    width={40}
  />
  <div
    className="text-white text-sm ml-2"
  >
    TestName
  </div>
</div>
`;
